### PR TITLE
chore(ui-react-native): Add default footer

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change adds a default footer for Authenticator sub components.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/Authenticator/common/DefaultFooter/DefaultFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { View } from 'react-native';
+
+import { DefaultFooterProps } from './types';
+
+export default function DefaultFooter({
+  children,
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/common/DefaultFooter/DefaultFooter.tsx:7]: Can we add `rest` props here?

**Review by joebuono — APPROVED** [review]

File: packages/react-native/src/Authenticator/Defaults/ConfirmResetPassword/ConfirmResetPassword.tsx [hunk 2]
```diff
@@ -2,17 +2,15 @@ import React from 'react';
 import { Text } from 'react-native';
 
 import { ConfirmResetPasswordComponent } from './types';
+import { DefaultFooter } from '../../common/DefaultFooter';
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmResetPassword/ConfirmResetPassword.tsx:5]: nit: Can we just do:

```jsx
import { DefaultFooter, DefaultHeader } from '../../common';
```

Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/Defaults/ConfirmResetPassword/ConfirmResetPassword.tsx:5]: I prefer to allow each export to be imported from its own module versus only through the main entry point (index file) as this reduces path resolution time.

File: packages/react-native/src/Authenticator/common/DefaultFooter/DefaultFooter.tsx (hunk 1 above)
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/common/DefaultFooter/DefaultFooter.tsx:7]: done!

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**